Updated data model

hylkevds opened the pull request:
Various data model updates required for CSAF modeling

Commit:
Fixed local installation scripts

Commit:
Update to V4.3.1 of netbox

Joerg322 committed:
Adjust model attributes to P625

eviceType:model_number
DeviceType:article_number
DeviceType:hardware_name
Device:year
Device:inventory_number
Software:manufacturer

Joerg322 committed:
Manufacturer Attribute of Software is a ForeignKey

Joerg322 committed:
new generation of device_type name

Joerg322 committed:
hardware_name in DeviceType Namen, Ip Addr /32 statt /24

Joerg322 committed:
Length of DeviceFinding:manufacturer

hylkevds committed:
Fixed typos

hylkevds committed:
Allow filtering of Software by manufacturer_id in API

hylkevds committed:
Make manufacturer name in Software list clickable

hylkevds committed:
Allow extra buttons in Software view

hylkevds committed:
Added migrations for model updates

Crubumble commented on scripts/configuration.py:
```diff
@@ -245,4 +245,4 @@
 
 # If true disables miscellaneous functionality which depends on access to the Internet.
-ISOLATED_DEPLOYMENT = _environ_get_and_map('ISOLATED_DEPLOYMENT', 'True', _AS_BOOL)
+#ISOLATED_DEPLOYMENT = _environ_get_and_map('ISOLATED_DEPLOYMENT', 'True', _AS_BOOL)
```
see comment other config file

Crubumble commented on scripts/install.bash:
Why the fix absolute path /home/ ?

Please use the `pwd=$(pwd)` as starting point and relative path of the script `cd "$(dirname "$0")"`

hylkevds commented:
Answer of @Joerg322: aktuelle kompatible Version bei Projektbeginn

Commit:
Add help for pull request

Crubumble commented:
> Answer of @Joerg322: aktuelle kompatible Version bei Projektbeginn

I am pleased to hear that you would like to voluntarily provide the functionality of the plugin under NetBox version 4.3.1. Since this is a newer version than the one in the main branch, please adjust the installation files such as Dockerfile in your pull request accordingly.

hylkevds committed:
Fixed #52: Software should be unique on name, manufacturer, version

hylkevds committed:
Fixed #53: Add manufacturer to software self-string

hylkevds committed:
Improved reliability of custom field registration

Instead of completely failing at the first problem, try to register each
custom field separately.

hylkevds committed:
Further improve reliability of custom-field creation

When upgrading from an older version of the plugin, newly
added custom fields were not created. This now always
checks all custom fields and creates the missing ones.

Crubumble committed:
Merge branch 'main' into P625-2

Crubumble commented:
Please check the former review comments and make a new PR ob branch integration/P625

hylkevds committed:
Updated netbox docker-image versions

hylkevds committed:
Updated netbox docker-image versions

hylkevds committed:
Ensure ordering on all models

See https://github.com/netbox-community/netbox/issues/20060

hylkevds committed:
Fixed custom fields not being updated after definition changes

Daniel-Rittershofer-Fraunhofer-IOSB committed:
fix: dynamic generation of device_type string
The device_type now ignores empty strings or "-" in its generation

Daniel-Rittershofer-Fraunhofer-IOSB committed:
fix: missing add button for FileHash in Navigation

Daniel-Rittershofer-Fraunhofer-IOSB committed:
fix #54, Fix navigation collapsing when adding items

Daniel-Rittershofer-Fraunhofer-IOSB committed:
fix: removed no longer needed get_absolute_url methods. fixes #30

hylkevds committed:
Fixed #56: Confusing button colours

hylkevds committed:
Set version number to 1.0

hylkevds commented:
The last commit is now tagged `V1.0 rc1`
